src/checkpoint/dispatch/vrt/virtual_serialize.h

```diff
@@ -157,21 +201,30 @@ struct SerializeAsVirtualIfNeeded<
  *     template <typename SerializerT>
  *     void serialize(SerializerT& s) {
  *       T* raw = elm.get();
- *       checkpoint::allocateConstructForPointer(s, raw);
+ *       checkpoint::reconstructPointedToObjectIfNeeded(s, raw);
  *       if (s.isUnpacking()) {
  *         a = std::shared_ptr<T>(raw);
  *       }
  *       s | *a;
  *     }
  *   };
  *
- *
  * \param[in] s the serializer
  * \param[in] target a reference to a pointer to the target object
  */
 template <typename SerializerT, typename T>
-void allocateConstructForPointer(SerializerT& s, T*& target) {
-  SerializeAsVirtualIfNeeded<T, SerializerT>::apply(s, target);
+void reconstructPointedToObjectIfNeeded(SerializerT& s, T*& target) {
```

Comment:
naming: is `reconstructPointeeIfNeeded` an option?

Reply:
I'd worry about it being misread as Pointer. I'll admit this is way more verbose than I'd like.

Reply:
ok, explicit is probably better than concise in the interface